### ISSUE TITLE
#:constructor-name + provide does not hide default constructor

### ISSUE THREAD

**Issue by bennn** [racket/typed-racket#312]
Typed Racket's `provide` revives the non-existent default constructor for `s`:

```
#lang racket/base

(module u racket/base
  (struct s () #:constructor-name make-s)
  (provide s))

(module t typed/racket/base
  (struct s () #:constructor-name make-s)
  (provide s))

;; Replacing with (require 'u) raises an exception --
;; the same exception you'd see for calling (s) inside module t
(require 't)
(s)
```

**Comment by bennn** [comment]
Hang on, my example is wrong but there is still a problem.

**Comment by takikawa** [comment]
The issue seems to be due to the line https://github.com/racket/typed-racket/blob/master/typed-racket-lib/typed-racket/typecheck/provide-handling.rkt#L95 which always assumes the struct type name is a constructor. Is that something we can make more precise now?

**Comment by bennn** [comment]
I think so, by making a Typed Racket version of `struct-info` like Sam suggests in #322

**Comment by capfredf** [comment]
This bug was fixed in #1051
